### PR TITLE
Distinguish CRIU crash from CRIU fail in ZDTM

### PR DESCRIPTION
**Pull request by sankalp-12**
This PR distinguishes `criu dump` crash from `criu dump` fail in the `zdtm.py` script. Now, *zdtm* reports `criu dump` crash as a test failure. 
Fixes the issue https://github.com/checkpoint-restore/criu/issues/350.

### CODE REVIEW

File: test/zdtm.py [hunk 1]
```diff
@@ -1233,8 +1233,7 @@ def __criu_act(self, action, opts=[], log=None, nowait=False):
                     return
             rst_succeeded = os.access(
                 os.path.join(__ddir, "restore-succeeded"), os.F_OK)
-            if self.__test.blocking() or (self.__sat and action == 'restore' and
-                                          rst_succeeded):
+            if (self.__test.blocking() and ret > 0) or (self.__sat and action == 'restore' and rst_succeeded):
```
Comment (mihalicyn) [inline test/zdtm.py:1236]: Looks correct. The only question is the case when `self.__criu == criu_rpc`, the `run` static method in this case can return negative value for any error, not related to SEGFAULT / SIGBUS. I can suggest to add a new static method (name it like `signal_exit`?) to `criu_rpc` and `criu_cli` to detect "signal exit" case. You can implement this method for `criu_cli` properly and for `criu_rpc` just always return `False` (and mark as `TODO`) for now.

Comment (sankalp-12) [reply test/zdtm.py:1248]: Done. Thanks for the suggestion!

File: test/zdtm.py [hunk 2]
```diff
@@ -1233,11 +1245,13 @@ def __criu_act(self, action, opts=[], log=None, nowait=False):
...
                 raise test_fail_expected_exc(action)
             else:
-                raise test_fail_exc("CRIU %s" % action)
+                if ((self.__criu == criu_cli and not self.__criu.signal_exit(ret)) or (self.__criu == criu_rpc and not self.__criu.signal_exit(ret))):
```
Comment (mihalicyn) [inline test/zdtm.py:1251]: this hunk looks strange. Why you need to check `self.__criu == criu_cli`?

Comment (sankalp-12) [reply test/zdtm.py:1251]: I checked it just as a precaution. Now, that I think about it, I don't think `criu_cli` would ever return a negative return code  for a non-signal exit.

File: test/zdtm.py (hunk 1 above)
Comment (mihalicyn) [reply test/zdtm.py:1248]: my idea was to move `ret > 0` check into `signal_exit` method. So, `if (self.__test.blocking() and ret > 0) or (self.__sat and action == 'restore' and rst_succeeded):` should be like `if (self.__test.blocking() and !self.__criu.signal_exit(ret)) or (self.__sat and action == 'restore' and rst_succeeded):`.

```
class criu_cli:
...
    @staticmethod
    def signal_exit(ret):
        return ret < 0
```

Reason is to keep all class-specific details in the class itself.

**Review by mihalicyn — CHANGES_REQUESTED** [review]

File: test/zdtm.py (hunk 2 above)
Comment (sankalp-12) [reply test/zdtm.py]: I will draft a new PR with all the changes. Closing this one.

Comment (mihalicyn) [reply test/zdtm.py]: you don't need to create a new PR every time. You can use `git rebase -i HEAD~N`, `git commit --amend` and `git push -f` to fix your changes.

Comment (sankalp-12) [reply test/zdtm.py]: The new PR is at https://github.com/checkpoint-restore/criu/pull/2140. Okay, I'll try not drafting a new one unnecessarily now onwards.

**Pull request closed without merging**